relayer/src/services/relayer/optimism.ts

```diff
@@ -97,19 +98,43 @@ export class OptimismRelayerService extends BaseRelayerService {
 
     if (!isWithdrawn) {
       // Finalize message via OptimismPortal (it calls the L1CrossDomainMessenger internally)
+      const OptimismPortalContract = await this.messenger.contracts.l1.OptimismPortal.connect(
+        this.l1RpcProvider,
+      );
+      const gasLimit = await OptimismPortalContract.estimateGas.finalizeWithdrawalTransaction(
```

Comment:
Is this https://docs.ethers.org/v5/api/contract/contract/#contract-estimateGas a common thing in other projects?

I'm more used to the syntax where you do provider.estimateGas(tx).